app/src/main/java/com/example/mytestproject/di/WeatherDataRepositoryModule.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.mytestproject.di
+
+import com.example.data.apiservice.WeatherDataApiService
+import com.example.data.mappers.WeatherDataMapper
+import com.example.domain.repositories.WeatherDataRepository
+import com.example.data.implementationRepo.WeatherDataRepositoryImpl
+import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+
+@Module
+class WeatherDataRepositoryModule {
```

Comment:
Let's try to merge classes WeatherDataModule and WeatherDataRepositoryModule into one WeatherDataModule class. And move this class from app module to data module. 
So, WeatherDataComponent will use it as a dagger module from another project module (data).